lib/data/external/AwsClient.js

```diff
@@ -43,23 +73,33 @@ class AwsClient {
         }
 
         if (!stream) {
-            return this._client.putObject(uploadParams, err => {
+            return this._client.putObject(uploadParams, (err, data) => {
                 if (err) {
-                    const log = createLogger(reqUids);
                     logHelper(log, 'error', 'err from data backend',
                       err, this._dataStoreName);
                     return callback(errors.InternalError
                       .customizeDescription('Error returned from ' +
                       `AWS: ${err.message}`)
                     );
                 }
-                return callback(null, awsKey);
+                if (!data.VersionId) {
+                    const error = errors.InternalError.customizeDescription(
+                        'Invalid state. Please ensure versioning is enabled ' +
+                        'in AWS for the location constraint\'s bucket and ' +
```

Comment:
I would just say for the 'location constraint' since we do allow location at the object level